src/main/java/duke/ParsedCommand.java

```diff
@@ -0,0 +1,109 @@
+package duke;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * Represents an input command that has been parsed into a class with accessible members.
+ */
+public class ParsedCommand {
+    private String commandType;
+    private int index;
+    private String name;
+    private String date;
+
+    /**
+     * Constructs a base ParsedCommand with the given commandType.
+     * @param commandType the intended action from the command
+     */
+    public ParsedCommand(String commandType) {
+        this.commandType = commandType;
+    }
+
+    /**
+     * Adds an index to this ParsedCommand.
+     * @param index the index to be saved to this ParsedCommand
+     * @return this modified ParseCommand
+     */
+    public ParsedCommand withIndex(int index) {
+        this.index = index;
+        return this;
+    }
+
+    /**
+     * Adds name to this ParsedCommand.
+     * @param name the name to be saved to this ParsedCommand
+     * @return this modified ParseCommand
+     */
+    public ParsedCommand withName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Adds a date to this ParsedCommand.
+     * @param date the index to be saved to this ParsedCommand
+     * @return this modified ParseCommand
+     */
+    public ParsedCommand withDate(String date) {
+        this.date = date;
+        return this;
+    }
```

Comment:
The main reason was to reduce the number of constructors. This way, even if future commands need different combinations of data fields, they can utilize the withX methods rather than creating new constructors.